Cargo.toml

```diff
@@ -14,11 +14,11 @@ edition = "2018"
 vendored = ["native-tls/vendored"]
 
 [dependencies]
-bytes = "0.5"
-native-tls = "0.2"
-hyper = { version = "0.13", default-features = false, features = ["tcp"] }
-tokio = { version = "0.2" }
-tokio-tls = "0.3"
+bytes = "1.0.0"
+native-tls = "0.2.6"
+hyper = { version = "0.14.1", default-features = false, features = ["tcp", "client", "http1", "http2"] }
```

Comment:
Is http2 actually necessary here? Or even http1? (Though possibly at least one of them?)

Reply:
This issue will be fixed by https://github.com/hyperium/hyper/issues/2376 @CryZe

Reply:
At least one of them is, sadly, or `hyper::client` won't be defined. `http2` also requires some special behaviour (ALPN negotiation, in particular), so it would be pretty confusing if you got stuck with HTTP/1.1 despite enabling `http2` on `hyper`.

That said, I can't seem to find where ALPN is handled here at the moment anyway...

Reply:
In hyper-rustls I only enabled `http1` since it was the lighter option of the two, but that wouldn't stop anybody from also enabling `http2`.

Reply:
@teozkr Seems like you don't need the http2 feature for ALPN: https://docs.rs/hyper/0.14.1/src/hyper/client/connect/mod.rs.html#175-178

(as you can see a bit further down only the `fn clone` has the feature gate)

iirc hyper-tls does not ALPN at all atm because not every native TLS crate supports it yet

Reply:
Doesn't look like we support HTTP/2 here anyway (at the moment), so I'll drop the `http2` feature.

Reply:
#61

Reply:
> @teozkr Seems like you don't need the http2 feature for ALPN: https://docs.rs/hyper/0.14.1/src/hyper/client/connect/mod.rs.html#175-178

That's pretty late though.. We probably shouldn't tell the server that we support HTTP/2 unless we actually do.

Reply:
Oh right, that's kinda even more painful to handle then, because you can't query which features are active for another crate, so you'd have to not only tell hyper that you want http2 but also tell the connector library about it :/

Reply:
Exactly.. :/

Well, disabled the `http2` feature for now anyway: 691c727

Reply:
`negotiate_h2` was already there in the previous hyper. The reason why it doesn't get called here I think is because native-tls doesn't support ALPN https://github.com/sfackler/rust-native-tls/issues/49

Reply:
Looks like that might be fixed soon: https://github.com/sfackler/rust-native-tls/pull/194

Reply:
We just published hyper v0.14.2, so it's now possible to enable `client` and access `hyper::client::connect` types without having to pick an HTTP proto feature.

Reply:
ca26cf9